### DISCUSSION TITLE
Any way to configure ddev to use localhost instead of ddev.site?

### DISCUSSION THREAD

**Discussion by brezanac** [ddev/ddev#6231] [Q&A] [answered]
Is there a way to configure ddev to use `localhost` instead of `ddev.test`?

In other words I'd like to use `project_name.localhost` instead of `project_name.ddev.test` to access my projects in the browser.

This would also help with developing features which require [secure browser context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts), considering that browsers trigger securew context for `localhost` but unfortunately not for domains such as `ddev.test`.

I've gone through ddev documentation and either I've missed the piece or relevant information or ddev doesn't support that option.

Thank you.

**Comment by penyaskito** [comment]
`ddev config --project-tld=localhost`

**Comment by stasadev** [comment]
Hi @brezanac,

You can do this, but it's not recommended because the contact between different projects using `*.localhost` URLs is broken:

- #6040

And currently there is no solution other than using a different TLD.

**Comment by brezanac** [comment]
> `ddev config --project-tld=localhost`

I have no idea how I missed this in the documentation. Thank you.

> Hi @brezanac,
> 
> You can do this, but it's not recommended because the contact between different projects using `*.localhost` URLs is broken:
> 
> * [Contact to `localhost` TLD broken in Debian Bookworm, curl: (60) SSL: no alternative certificate subject name #6040](https://github.com/ddev/ddev/issues/6040)
> 
> And currently there is no solution other than using a different TLD.

That's unfortunate because I absolutely need secure context to work for my projects. That or having to go through the hassle of setting up HTTPS etc.

Thank you.

**Reply by rfay** [reply]
You mention `ddev.test`, which is not the default that DDEV uses; it uses `ddev.site`, which is far more likely to work for you; `test` and `localhost` are both "suspect" TLDs in the scheme of things.

**Comment by brezanac** [comment]
> You mention `ddev.test`, which is not the default that DDEV uses; it uses `ddev.site`, which is far more likely to work for you; `test` and `localhost` are both "suspect" TLDs in the scheme of things.

Sorry, I actually meant the default TLD which comes preconfigured with ddev (`ddev.site`) not `ddev.test`.

Also, `localhost` has a very special meaning for browsers because it's considered to "run" in a [Secure Context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts).

Additionally, some of the browser APIs are only available in secure context.

[Features restricted to secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts/features_restricted_to_secure_contexts)

Secure context can be easily checked in the browser dev tools by running `window.isSecureContext`.

The `site` domain on the other hand has no significance for browsers as it's considered out of secure context unless served over HTTPS or WSS.

**Reply by rfay** [reply, marked as answer]
Yes... but `*.ddev.site` is almost always served over HTTPS in DDEV, unless something is misconfigured. Our aim with DDEV is to provide an environment as close to real life (with HTTPS, for example) as possible

**Reply by penyaskito** [reply]
Just tested this and confirm that in a regular ddev site with no special config, using ddev.site as tld, `window.isSecureContext` shows `true`.

**Reply by brezanac** [reply]
HTTPS is most definitely a secure context, however it seems that there's an issue with `mkcert` on my machine when ddev tries to register the CA, which results in an HTTP launch URL once ddev runs everything.

I am pretty sure the issue isn't tied to ddev itself but if it is I'll probably open a new issue here.

Thanks again for the insight and additional info.

**Reply by rfay** [reply]
I'd be happy to help you sort out your mkcert situation. Come on over to [Discord](https://discord.gg/5wjP76mBJD) and provide your situation with a `ddev debug test` and we can chase it down. It's solvable everywhere.
